cmd/notary-server/config.go

```diff
@@ -21,9 +23,21 @@ import (
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/spf13/viper"
 	"golang.org/x/net/context"
-	"gopkg.in/dancannon/gorethink.v2"
+	gorethink "gopkg.in/dancannon/gorethink.v2"
 )
 
+// gets the required gun prefixes accepted by this server
+func getRequiredGunPrefixes(configuration *viper.Viper) ([]string, error) {
+	prefixes := configuration.GetStringSlice("repositories.gun_prefixes")
+	for _, prefix := range prefixes {
+		p := path.Clean(strings.TrimSpace(prefix))
+		if p+"/" != prefix || strings.HasPrefix(p, "/") || strings.HasPrefix(p, "..") {
```

Comment:
I was wondering that too - https://github.com/docker/notary/issues/742 tracks the issue, but we can totally check it here.  I think @endophage is testing out whether the Go http server automatically collapses the URL for you too (e.g. whether the Go server collapses `http://localhost:8080/foo/../../../bar` to `http://localhost:8080/bar`)